### PR TITLE
TrailingCommas: add the `multiple` option

### LINKED ISSUE #1410: trailing commas "always" does not respect function arity
This template is a guideline, not a strict requirement.
- **Version**: 2.0.0-RC6
- **Integration**: IntelliJ
- **Configuration**:
```
version = 2.0.0-RC6

align = true

maxColumn = 120

rewrite.rules = [RedundantBraces]

rewrite.redundantBraces.stringInterpolation = true

newlines.alwaysBeforeMultilineDef = true

newlines.alwaysBeforeTopLevelStatements = true

trailingCommas = always

```

## Steps

Given code like this:
```scala
   case Scrolled(key) =>
      resultFor(key) { rts =>
        rts.viewport.fold(noChange) { v =>
          val l = v.top(rts)
          effectOnly(
            Effect.action(
              ActionBatch(
                UpdateReaderState(rts.copy(scrollY = v.scrollY, location = l)),
                FindPageNumber(key)
              )
            )
          )
        }
      }
```

When I run scalafmt like this:

```bash
scalafmt
```

## Problem

Scalafmt formats code like this:
```scala
    case Scrolled(key) =>
      resultFor(key) { rts =>
        rts.viewport.fold(noChange) { v =>
          val l = v.top(rts)
          effectOnly(
            Effect.action(
              ActionBatch(
                UpdateReaderState(rts.copy(scrollY = v.scrollY, location = l)),
                FindPageNumber(key),
              ),
            ),
          )
        }
      }
```

## Expectation

I would like the formatted output to look like this:
```scala
    case Scrolled(key) =>
      resultFor(key) { rts =>
        rts.viewport.fold(noChange) { v =>
          val l = v.top(rts)
          effectOnly(
            Effect.action(
              ActionBatch(
                UpdateReaderState(rts.copy(scrollY = v.scrollY, location = l)),
                FindPageNumber(key),
              )
            )
          )
        }
      }
```

## Workaround

I haven't found any workaround.
## Notes

It seems like trailing commas "always" configuration should respect function/method arity - otherwise it looks like every function/ethod is a var-arg function/method.

### PR DESCRIPTION
**Pull request by kitbellew**
Fixes #1410.

### CODE REVIEW

**Comment by kitbellew** [comment]
@olafurpg @tanishiking @poslegm PTAL

**Comment by kitbellew** [comment]
@neko-kai FYI

**Comment by neko-kai** [comment]
@kitbellew Thank you! 👍

File: scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala [hunk 1]
```diff
@@ -373,6 +381,24 @@ class FormatWriter(formatOps: FormatOps) {
                 if !tok.left.is[T.Comma] && isClosedDelimWithNewline(true) =>
               sb.append(',')
               ws(-1)
+            // append comma if newline and multiple args
+            case TrailingCommas.multiple
+                if !tok.left.is[T.Comma] &&
+                  getClosedDelimWithNewline(true).exists { ft =>
```
Comment (poslegm) [inline scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala]: May be extract it to a method?

Comment (kitbellew) [reply scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatWriter.scala]: will do, thank you

**Pull request merged**